driver-core/src/main/com/mongodb/connection/SnappyCompressor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.connection;
+
+import com.mongodb.MongoInternalException;
+import org.bson.ByteBuf;
+import org.bson.io.BsonOutput;
+import org.xerial.snappy.Snappy;
+import org.xerial.snappy.SnappyInputStream;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+class SnappyCompressor extends Compressor {
+    @Override
+    public String getName() {
+        return "snappy";
+    }
+
+    @Override
+    public byte getId() {
+        return 1;
+    }
+
+    // the server does not support the framing format so SnappyFramedOutputStream can't be used.  The entire source message must first
```

Comment:
Did you add a ticket about framing support? Or is it something that will never be supported?